test/Microsoft.ML.Tests/TrainerEstimators/FAFMEstimator.cs

```diff
@@ -2,15 +2,43 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Linq;
 using Microsoft.ML.Data;
 using Microsoft.ML.FactorizationMachine;
 using Microsoft.ML.RunTests;
+using Microsoft.ML.SamplesUtils;
 using Xunit;
 
 namespace Microsoft.ML.Tests.TrainerEstimators
 {
     public partial class TrainerEstimators : TestDataPipeBase
     {
+        [Fact]
+        public void FfmBinaryClassificationWithAdvancedArguments()
+        {
+            var mlContext = new MLContext(seed: 0);
+            var data = DatasetUtils.GenerateFfmSamples(500);
+            var dataView = ComponentCreation.CreateDataView(mlContext, data.ToList());
+
+            var ffmArgs = new FieldAwareFactorizationMachineTrainer.Arguments();
+
+            // Customized the field names.
+            ffmArgs.FeatureColumn = nameof(DatasetUtils.FfmExample.Field0); // First field.
+            ffmArgs.ExtraFeatureColumns = new[]{ nameof(DatasetUtils.FfmExample.Field1), nameof(DatasetUtils.FfmExample.Field2) };
```

Comment:
>ExtraFeatureColumns [](start = 20, length = 19)

this looks slightly odd . isn't it ?

am curious -- why move away from the convention used in iteration #2, where  we were re-defining  `FeatureColumn`  as a string[]

Reply:
Framework of generating entry points generates the hidden field (old feature column name) and therefore we have two fields with the same name and an error.

---
In reply to: [250309618](https://github.com/dotnet/machinelearning/pull/2205#discussion_r250309618) [](ancestors = 250309618)